modules/internal/ArrayViewRankChange.chpl

```diff
@@ -22,9 +22,337 @@
+      } else {
+        for i in downdom do
+          yield downIdxToUpIdx(i);
+      }
+    }
+
+    iter these(param tag: iterKind) where tag == iterKind.standalone {
+      if chpl__isDROrDRView(downdom) {
+        for i in updom.these(tag) do
+          yield i;
+      } else {
+        for i in downdom.these(tag) do
+          yield downIdxToUpIdx(i);
+      }
+    }
+
+    iter these(param tag: iterKind) where tag == iterKind.leader {
+      if chpl__isDROrDRView(downdom) {
+        for followThis in updom.these(tag) do
+          yield followThis;
+      } else {
+        for followThis in downdom.these(tag) {
+          const followThisLoD = chpl_rankChangeConvertHiDTupleToLoD(followThis);
+          yield followThisLoD;
+        }
+      }
+    }
+
+    iter these(param tag: iterKind, followThis)
+      where tag == iterKind.follower {
+      if chpl__isDROrDRView(downdom) {
+        for i in updom.these(tag, followThis) do
+          yield i;
+      } else {
+        const followThisHiD = chpl_rankChangeConvertLoDTupleToHiD(followThis);
+        for i in downdom.these(tag, followThisHiD) {
+          yield chpl_rankChangeConvertIdxHiDToLoD(i, collapsedDim, idx, rank);
+        }
+      }
+    }
+
+    proc chpl_rankChangeConvertLoDTupleToHiD(tup) {
+      var tupHiD: downrank*tup(1).type;
+      var i = 1;
+      for param d in 1..downrank do
+        if collapsedDim(d) then
+          tupHiD(d) = 0..0;  // TODO: This seems bogus/arbitrary... or is it?
+        else {
+          tupHiD(d) = tup(i);
+          i += 1;
+        }
+      return tupHiD;
+    }
+
+    proc chpl_rankChangeConvertHiDTupleToLoD(tup) {
+      var tupLoD: rank*tup(1).type;
+      var i = 1;
+      for param d in 1..downrank do
+        if !collapsedDim(d) {
+          tupLoD(i) = tup(d);
+          i += 1;
+        }
+      return tupLoD;
+    }
+
+
+    inline proc downIdxToUpIdx(downIdx) {
+      var upIdx: rank*idxType;
+      var upDim = 1;
+      for param downDim in 1..downrank {
+        if !collapsedDim(downDim) {
+          upIdx(upDim) = downIdx(downDim);
+          upDim += 1;
+        }
+      }
+      if rank == 1 then
+        return upIdx(1);
+      else
+        return upIdx;
+    }
+
+    // TODO: Is there something we can re-use here?
+    proc dsiSerialWrite(f) {
+      var first = true;
+      for d in 1..downrank do
+        if !collapsedDim(d) {
+          if first {
+            write("{");
+            first = false;
+          } else
+            write(", ");
+          write(downdom.dsiDim(d));
+        }
+      write("}");
+    }
+
+    proc dsiMyDist() {
+      return dist;
+    }
+
+    proc dsiTargetLocales() {
+      //
+      // BLC: To tighten this up, we'd need to query the distribution to
+      // see what subset of target locales the rank-change slice hit vs.
+      // not.
+      //
+      //      compilerWarning("Calls to .targetLocales() on rank-change slices may currently return a superset of the locales targeted.");
```

Comment:
I added it in the original array view work, but commented it out here because on reflection I didn't think it was all that helpful.  Slicing may similarly restrict which locales own a piece of the view, but doesn't have a similar warning.  In the case of privatizing a distributed array, I also wasn't certain whether the answer to this query should be "everyone who owns a privatized copy of the array's class" or "only those locales that own non-zero data."  For complex distributions, it also seemed as though it might be challenging/expensive to determine exactly which target locales own a slice.

Mostly, I felt dissatisfied with the definition of targetLocales and didn't feel confident that preserving the warning here had value.